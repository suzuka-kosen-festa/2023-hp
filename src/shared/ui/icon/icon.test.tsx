import { composeStory } from "@storybook/react";
import { render } from "@testing-library/react";
import Meta, {
  Cloud1,
  Cloud2,
  Cloud3,
  Cloud4,
  Cloud5,
  Defect,
  LogoStatic,
} from "./icon.stories";

describe("[shared/ui/icon] icon.tsx", () => {
  it("should render the defect component", () => {
    const Composed = composeStory(Defect, Meta);
    const { container } = render(<Composed />);
    expect(container).toMatchSnapshot();
  });
  it("should render the cloud1 component", () => {
    const Composed = composeStory(Cloud1, Meta);
    const { container } = render(<Composed />);
    expect(container).toMatchSnapshot();
  });
  it("should render the cloud2 component", () => {
    const Composed = composeStory(Cloud2, Meta);
    const { container } = render(<Composed />);
    expect(container).toMatchSnapshot();
  });
  it("should render the cloud3 component", () => {
    const Composed = composeStory(Cloud3, Meta);
    const { container } = render(<Composed />);
    expect(container).toMatchSnapshot();
  });
  it("should render the cloud4 component", () => {
    const Composed = composeStory(Cloud4, Meta);
    const { container } = render(<Composed />);
    expect(container).toMatchSnapshot();
  });
  it("should render the cloud5 component", () => {
    const Composed = composeStory(Cloud5, Meta);
    const { container } = render(<Composed />);
    expect(container).toMatchSnapshot();
  });
  it("should render the logoStatic component", () => {
    const Composed = composeStory(LogoStatic, Meta);
    const { container } = render(<Composed />);
    expect(container).toMatchSnapshot();
  });
});
