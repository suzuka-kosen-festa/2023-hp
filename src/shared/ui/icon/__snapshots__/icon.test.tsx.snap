// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[shared/ui/icon] icon.tsx > should render the cloud1 component 1`] = `
<div>
  <span
    class="styles_icon__1jklp5o0 styles__1jklp5o1 styles_iconType_type_cloud1__1jklp5o3"
  />
</div>
`;

exports[`[shared/ui/icon] icon.tsx > should render the cloud2 component 1`] = `
<div>
  <span
    class="styles_icon__1jklp5o0 styles__1jklp5o1 styles_iconType_type_cloud2__1jklp5o4"
  />
</div>
`;

exports[`[shared/ui/icon] icon.tsx > should render the cloud3 component 1`] = `
<div>
  <span
    class="styles_icon__1jklp5o0 styles__1jklp5o1 styles_iconType_type_cloud3__1jklp5o5"
  />
</div>
`;

exports[`[shared/ui/icon] icon.tsx > should render the cloud4 component 1`] = `
<div>
  <span
    class="styles_icon__1jklp5o0 styles__1jklp5o1 styles_iconType_type_cloud4__1jklp5o6"
  />
</div>
`;

exports[`[shared/ui/icon] icon.tsx > should render the cloud5 component 1`] = `
<div>
  <span
    class="styles_icon__1jklp5o0 styles__1jklp5o1 styles_iconType_type_cloud5__1jklp5o7"
  />
</div>
`;

exports[`[shared/ui/icon] icon.tsx > should render the defect component 1`] = `
<div>
  <span
    class="styles_icon__1jklp5o0 styles__1jklp5o1 styles_iconType_type_defect__1jklp5o2"
  />
</div>
`;

exports[`[shared/ui/icon] icon.tsx > should render the logoStatic component 1`] = `
<div>
  <span
    class="styles_icon__1jklp5o0 styles__1jklp5o1 styles_iconType_type_logoStatic__1jklp5o8"
  />
</div>
`;
